primitiv/c/graph.cc

```diff
@@ -7,224 +7,193 @@
 #include <vector>
 
 #include <primitiv/graph.h>
-
 #include <primitiv/c/internal.h>
 #include <primitiv/c/graph.h>
 
 using primitiv::Node;
 using primitiv::Graph;
+using primitiv::c::internal::to_c_ptr;
+using primitiv::c::internal::to_cpp_ptr;
+using primitiv::c::internal::to_c_ptr_from_value;
 
 extern "C" {
 
 primitiv_Node *primitiv_Node_new() {
-  return to_c(new Node);
-}
-primitiv_Node *safe_primitiv_Node_new(primitiv_Status *status) {
-  SAFE_RETURN(primitiv_Node_new(), status, nullptr);
+  return to_c_ptr(new Node);
 }
 
-primitiv_Node *primitiv_Node_new_with_movement(primitiv_Node *node) {
-  return to_c(new Node(std::move(*to_cc(node))));
-}
-primitiv_Node *safe_primitiv_Node_new_with_movement(primitiv_Node *node,
-                                                    primitiv_Status *status) {
-  SAFE_RETURN(primitiv_Node_new_with_movement(node), status, nullptr);
+primitiv_Status primitiv_Node_new_from_node(primitiv_Node *src,
```

Comment:
This function looks dangerous; it is hard (probably not able) to expect an implicit **moving** behavior in the function by looking the name `primitiv_Node_new_from_node` (most users may expect that this function **copies** the `Node`).
I think that any *move semantics* should not be implemented by C APIs, and all move semantics in other languages should be implemented by moving pointer values.